### PR TITLE
Ensure error logs are always printed

### PR DESCRIPTION
**Pull request by FelonEkonom**

### CODE REVIEW

File: lib/membrane/core/element.ex [hunk 1]
```diff
@@ -198,7 +209,9 @@ defmodule Membrane.Core.Element do
       :erlang.process_info(self(), :message_queue_len) |> elem(1)
     )
 
-    do_handle_info(message, state)
+    Macros.log_on_error do
+      do_handle_info(message, state)
+    end
```
Comment (varsill) [inline lib/membrane/core/element.ex]: why `Telemetry.report_metric` is outside of the `Macros.log_on_error`?

File: lib/membrane/core/macros.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
+defmodule Membrane.Core.Macros do
+  @moduledoc false
+
+  # For some reason GenServer processes sometimes don't print logs about crash, so
+  # we add this macro, to ensure that error logs wil be always logged
```
Comment (varsill) [inline lib/membrane/core/macros.ex:5]:
```suggestion
  # we add this macro, to ensure that error logs are always printed
```

File: lib/membrane/core/macros.ex [hunk 3]
```diff
@@ -0,0 +1,24 @@
+defmodule Membrane.Core.Macros do
```
Comment (mat-hek) [inline lib/membrane/core/macros.ex:1]:
```suggestion
defmodule Membrane.Core.Utils do
```

**Review by mat-hek — APPROVED** [review]

**Review by varsill — CHANGES_REQUESTED** [review]
How about adding it to other GenServers that we define in Core as well (like ResourceGuard etc.?)

**Review by varsill — APPROVED** [review]
Initially I just thought of adding it only in other processes that can execute users code (just like `ResourceGuard`, which executes registered user's lambda), but since you have added logging to all GenServers it's even better, as it will be easier for us to debug if we made some mistakes :D

**Pull request merged**